### PR TITLE
Mv3

### PR DESCRIPTION
**Pull request by jeremyriverain**
Hello,

In order to better understand the implications of the new manifest V3, I made the effort to migrate from version 2 to version 3. 
Perhaps this work is a bit premature, given the recent publication of the new manifest. 
Anyway, I post here the result of my work, in case you are interested in some parts.

Regards.
Jérémy Riverain

### CODE REVIEW

**Comment by jeremyriverain** [comment]
@googlebot I signed it!

**Comment by rviscomi** [comment]
Thank you for working on this @jeremyriverain! Sorry for the delay. If you're still interested, could you sync to the latest version and resolve the merge conflicts? If not I'm happy to pick up any remaining work on this PR. It'd be great to complete the Manifest v3 migration soon, as I understand we're [running out of time](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/).

**Comment by jeremyriverain** [comment]
@rviscomi sure i can work on it. It would be a pleasure to contribute. 

Just give me a few days to find a moment and understand the changes to the repository since the last time.

**Comment by jeremyriverain** [comment]
I rebased this branch on the master branch. There were only minor conflicts in the `manifest.json` file.

It seems that the extension works as before. That said, as I'm not the maintainer of the repo, I can't be sure.

Can you look at this work and tell me what you think please ? Is there anything I can do ?

**Comment by rviscomi** [comment]
Thank you @jeremyriverain! I've tested your branch locally and the extension appears to be working correctly. I'll go through the code more closely and leave any feedback.

File: manifest.json [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
   "options_page": "src/options/options.html",
-  "permissions": [
-    "tabs",
-    "storage",
-    "activeTab",
+  "host_permissions": [
     "http://localhost/",
     "http://*/*",
     "https://*/*",
     "https://www.googleapis.com/pagespeedonline/v5/runPagespeed?*"
```
Comment (rviscomi) [inline manifest.json:22]: Can we collapse everything into this single pattern?

https://developer.chrome.com/docs/extensions/mv3/match_patterns/

```suggestion
    "*://*/*"
```

File: manifest.json [hunk 2]
```diff
@@ -37,8 +40,9 @@
     }
   ],
   "background": {
-    "scripts": ["src/bg/background.js"],
-    "persistent": false
+    "service_worker": "service_worker.js"
   },
-  "content_security_policy": "script-src 'self' https://www.googleapis.com; object-src 'self'"
+  "content_security_policy": {
+    "extensions": "script-src 'self' https://www.googleapis.com; object-src 'self'"
```
Comment (rviscomi) [inline manifest.json:46]: A few of changes:

- the property should be `extension_pages` per https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#content-security-policy
- use the `default-src 'self'` catch-all directive
- use the CrUX-specific host for the `connect-src` directive

```suggestion
    "extension_pages": "default-src 'self'; connect-src https://chromeuxreport.googleapis.com;"
```

File: manifest.json [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
-    "src/browser_action/viewer.css",
-    "src/browser_action/web-vitals.js",
-    "/node_modules/web-vitals/dist/web-vitals.js"
+    {
+      "resources": [
+        "src/browser_action/viewer.css",
+        "src/browser_action/web-vitals.js",
+        "/node_modules/web-vitals/dist/web-vitals.js"
+      ],
+      "matches": ["https://*/*", "http://*/*"]
```
Comment (rviscomi) [inline manifest.json:32]:
```suggestion
      "matches": ["*://*/*"]
```

File: manifest.json [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
+  "permissions": ["tabs", "storage", "activeTab", "scripting"],
   "web_accessible_resources": [
-    "src/browser_action/viewer.css",
-    "src/browser_action/web-vitals.js",
-    "/node_modules/web-vitals/dist/web-vitals.js"
+    {
+      "resources": [
+        "src/browser_action/viewer.css",
+        "src/browser_action/web-vitals.js",
+        "/node_modules/web-vitals/dist/web-vitals.js"
```
Comment (rviscomi) [inline manifest.json:30]: Doesn't seem like we're using the `node_modules` version for now, but we should probably fix that soon.

```suggestion
        "src/browser_action/web-vitals.js"
```

**Review by rviscomi — CHANGES_REQUESTED** [review]

**Comment by rviscomi** [comment]
Woohoo thanks @jeremyriverain! I changed the destination branch from `master` to a new one named `manifest-v3`. We just released v1.1.1 to the store yesterday, and I'm anticipating that we may need to push some higher priority bug fixes in the coming weeks. Decoupling these changes from `master` lets us publish bug fixes without risking more issues caused by the v3 migration.

**Review by rviscomi — APPROVED** [review]

**Pull request merged**